index.js

```diff
@@ -38,12 +39,21 @@ Cloneable.prototype.clone = function () {
   }
 
   this._clonesCount++
-  return pump(this, new Clone(this))
+  return this.pipe(new Clone(this))
+}
+
+function forwardDestroy (src, dest) {
+  src.on('error', destroy)
+  src.on('close', destroy)
+
+  function destroy (err) {
+    dest.destroy(err)
+  }
```

Comment:
we should use https://github.com/mafintosh/end-of-stream to check that the stream errors